src/dbnode/storage/series/types.go

```diff
@@ -133,15 +144,7 @@ type DatabaseSeries interface {
 	Close()
 
 	// Reset resets the series for reuse.
-	Reset(
-		id ident.ID,
-		tags ident.Tags,
-		uniqueIndex uint64,
-		blockRetriever QueryableBlockRetriever,
-		onRetrieveBlock block.OnRetrieveBlock,
-		onEvictedFromWiredList block.OnEvictedFromWiredList,
-		opts Options,
-	)
+	Reset(opts DatabaseSeriesOptions)
```

Comment:
Would it be a good idea to change Reset to return an error, e.g. if some of the necessary fields in `DatabaseSeriesOptions` are unset?

Reply:
Maybe we do this as a follow up?  It has wide reaching implications unfortunately due to the amount of callers.